### ISSUE TITLE
Every few hours `err="unable to delete version 14371500 with 1 active readers"`

### ISSUE THREAD

**Issue by kucharskim** [cosmos/gaia#2272]
## Summary of Bug

RPC nodes fails every few hours with:

```
gaia_1  | 3:11AM ERR CONSENSUS FAILURE!!! err="unable to delete version 14371500 with 1 active readers" module=consensus stack="goroutine 12429 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:24 +0x65\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine.func2()\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:724 +0x4c\npanic({0x1a3d260, 0xc0819028e8})\n\truntime/panic.go:838 +0x207\ngithub.com/cosmos/cosmos-sdk/store/rootmulti.(*Store).PruneStores(0xc00189e820, 0x1, {0x0?, 0x0?, 0x25730a8?})\n\tgithub.com/cosmos/cosmos-sdk@v0.45.12/store/rootmulti/store.go:454 +0x285\ngithub.com/cosmos/cosmos-sdk/store/rootmulti.(*Store).Commit(0xc00189e820)\n\tgithub.com/cosmos/cosmos-sdk@v0.45.12/store/rootmulti/store.go:423 +0x18b\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Commit(0xc0002e8a80)\n\tgithub.com/cosmos/cosmos-sdk@v0.45.12/baseapp/abci.go:317 +0x1e6\ngithub.com/tendermint/tendermint/abci/client.(*localClient).CommitSync(0xc000f6e9c0)\n\tgithub.com/tendermint/tendermint@v0.34.24/abci/client/local_client.go:264 +0xb6\ngithub.com/tendermint/tendermint/proxy.(*appConnConsensus).CommitSync(0x2a?)\n\tgithub.com/tendermint/tendermint@v0.34.24/proxy/app_conn.go:93 +0x22\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).Commit(_, {{{0xb, 0x0}, {0x25491e8, 0x8}}, {0xc0013c0010, 0xb}, 0x4f5b97, 0xdb4b1a, {{0xc0d2bf3620, ...}, ...}, ...}, ...)\n\tgithub.com/tendermint/tendermint@v0.34.24/state/execution.go:228 +0x269\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x0}, {0x25491e8, 0x8}}, {0xc0013c0010, 0xb}, 0x4f5b97, 0xdb4b1a, {{0xc0d2bf3620, ...}, ...}, ...}, ...)\n\tgithub.com/tendermint/tendermint@v0.34.24/state/execution.go:180 +0x6ee\ngithub.com/tendermint/tendermint/consensus.(*State).finalizeCommit(0xc000edd500, 0xdb4b1a)\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:1654 +0xafd\ngithub.com/tendermint/tendermint/consensus.(*State).tryFinalizeCommit(0xc000edd500, 0xdb4b1a)\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:1563 +0x2ff\ngithub.com/tendermint/tendermint/consensus.(*State).enterCommit.func1()\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:1498 +0x87\ngithub.com/tendermint/tendermint/consensus.(*State).enterCommit(0xc000edd500, 0xdb4b1a, 0x0)\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:1536 +0xcb7\ngithub.com/tendermint/tendermint/consensus.(*State).addVote(0xc000edd500, 0xc110fff2c0, {0xc001a3c000, 0x28})\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:2150 +0xb7c\ngithub.com/tendermint/tendermint/consensus.(*State).tryAddVote(0xc000edd500, 0xc110fff2c0, {0xc001a3c000?, 0xc112899f00?})\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:1948 +0x2c\ngithub.com/tendermint/tendermint/consensus.(*State).handleMsg(0xc000edd500, {{0x2550220?, 0xc11decea48?}, {0xc001a3c000?, 0x0?}})\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:853 +0x44b\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine(0xc000edd500, 0x0)\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:760 +0x419\ncreated by github.com/tendermint/tendermint/consensus.(*State).OnStart\n\tgithub.com/tendermint/tendermint@v0.34.24/consensus/state.go:379 +0x12d\n"
gaia_1  | 3:11AM INF service stop impl={"Logger":{}} module=consensus msg={} wal=/gaia/.gaiad/data/cs.wal/wal
gaia_1  | 3:11AM INF service stop impl={"Dir":"/gaia/.gaiad/data/cs.wal","Head":{"ID":"XizXeRwhDyIO:/gaia/.gaiad/data/cs.wal/wal","Path":"/gaia/.gaiad/data/cs.wal/wal"},"ID":"group:XizXeRwhDyIO:/gaia/.gaiad/data/cs.wal/wal","Logger":{}} module=consensus msg={} wal=/gaia/.gaiad/data/cs.wal/wal
```

## Version

```
# docker exec -ti 0bba2a710bb8 gaiad --home /tmp/foobar version
v8.0.0
```
our own images have also:
```
# docker exec -ti 0bba2a710bb8 cat /build-info
REPO_URL=https://github.com/cosmos/gaia.git
REPO_BRANCH=build_v8.0.0
REPO_HASH=f9507c7e79f258aaa9a2265ca7d94e53988ffe16
REPO_TAG=v8.0.0
```

## Steps to Reproduce

```
# cat docker-compose.yml
version: '3.4'
services:
  gaia:
    command: sh -c "rm -vf /gaia/.gaiad/kms.sock; exec gaiad --home /gaia/.gaiad start --x-crisis-skip-assert-invariants"
    hostname: gaia
    user: "30004:30004"
    image: gaia:v8.0.0
    restart: unless-stopped
    env_file:
      - env-statesync
    ports:
      - 26656:26656
      - 10.XXX.XXX.XXX:26657:26657
      - 10.XXX.XXX.XXX:26660:26660
      - 10.XXX.XXX.XXX:26617:1317
      - 10.XXX.XXX.XXX:26690:9090
    volumes:
        - /data/gaia:/gaia/.gaiad
    logging:
      options:
        max-size: "10g"
        max-file: "3"
```

```
# cat env-statesync 
GAIAD_STATESYNC_ENABLE=true
GAIAD_STATESYNC_RPC_SERVERS=http://10.XXX.XXX.XXX:26657,MORE_RPC1,MORE_RPC2,https://rpc.cosmos.network:443
GAIAD_STATESYNC_TRUST_HEIGHT=HEIGHT_FROM_RPC_FROM_OUR_AUTOMATION
GAIAD_STATESYNC_TRUST_HASH=HASH_RROM_RPC_GENERATED_BY_OUR_AUTOMATION
```

```
custom-gen-env-statesync-command
docker-compose down
docker-compose run gaia sh -c 'rm -fr /gaia/.gaiad/data/*/'
docker-compose down
docker-compose up --force-recreate -d
```
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

**Comment by kucharskim** [comment]
Logs from docker from `gaiad` at https://generic.mp/r/5mjb-khyu-kvu6-nygn-vdiz-im2d-zwqq/

**Comment by mmulji-ic** [comment]
Hi @kucharskim thanks for reporting. Ideally we'd like everyone to move to v8.0.1.
But before you do that, a validator mentioned that it may be related to pruning. 
See [issue](https://github.com/classic-terra/core/issues/122). 

Can you share your current pruning settings and if they are not pruning = "default", then try that.

**Comment by vchong** [comment]
The custom settings used that caused this issue was:
```
pruning = "custom"
pruning_keep_recent = "100"
pruning_keep_every = "0"
pruning_interval = "10"
```

**Comment by mmulji-ic** [comment]
@kucharskim after Telegram convo, feedback was that the issue is resolved with pruning set to default. Will open another issue to update docs to show what the valid combinations for pruning are and potential error messages that can be received. So that if encountered in the future, issue can be more quickly resolved.

If the same issue does reappear, please reopen this issue.

**Comment by vchong** [comment]
> Will open another issue to update docs to show what the valid combinations for pruning are and potential error messages that can be received.

@mmulji-ic Can you please link the issue here so we can follow the progress?

**Comment by mmulji-ic** [comment]
@vchong I got feedback from the SDK team, that this is fixed in SDK#46. We will move to SDK#47 later this year.
